### PR TITLE
Check for min/max set values

### PR DESCRIPTION
**Pull request by dieris**
Freq., power, and phase allowed values.

### CODE REVIEW

File: src/auspex/instruments/holzworth.py [hunk 1]
```diff
@@ -73,23 +73,35 @@ def frequency(self):
         return float(v.split()[0])*1e-3
     @frequency.setter
     def frequency(self, value):
-        self.ch_query(":FREQ:{} GHz".format(value))
+        fmin = float((self.ch_query(":FREQ:MIN?")).split()[0])
```
Comment (caryan) [inline src/auspex/instruments/holzworth.py]: is it expensive to read this on every set?

Comment (dieris) [inline src/auspex/instruments/holzworth.py]: I guess so, each query takes 4 ms. What if we set `self.fmin`, `self.fmax`... upon connection?

File: src/auspex/instruments/holzworth.py [hunk 2]
```diff
@@ -73,23 +73,35 @@ def frequency(self):
...
+        fmax = float((self.ch_query(":FREQ:MAX?")).split()[0])
+        if fmin*1e-3 <= value <= fmax*1e-3:
+            self.ch_query(":FREQ:{} GHz".format(value))
+        else:
+            err_msg = "The value {} GHz is outside of the allowable range {}-{} GHz specified for instrument '{}'.".format(value, fmin*1e-3, fmax*1e-3, self.name)
```
Comment (caryan) [inline src/auspex/instruments/holzworth.py]: I'm not we we want to require Python 3.6 but the [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings) are much nicer for this.

File: src/auspex/instruments/holzworth.py (hunk 1 above)
Comment (caryan) [inline src/auspex/instruments/holzworth.py]: yeah I think its better to cache on connect

File: src/auspex/instruments/holzworth.py (hunk 2 above)
Comment (blakejohnson) [inline src/auspex/instruments/holzworth.py]: I wouldn't introduce that dependency at this point. We're still running Python 3.5 in several places.

**Pull request merged**